src/org/loklak/api/amazon/AmazonProductService.java

```diff
@@ -0,0 +1,160 @@
+ *  
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *  
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program in the file lgpl21.txt
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.loklak.api.amazon;
+
+import java.io.StringWriter;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.json.JSONObject;
+import org.json.XML;
+import org.loklak.server.APIException;
+import org.loklak.server.APIHandler;
+import org.loklak.server.AbstractAPIHandler;
+import org.loklak.server.Authorization;
+import org.loklak.server.BaseUserRole;
+import org.loklak.server.Query;
+import org.loklak.tools.storage.JSONObjectWithDefault;
+import org.w3c.dom.Document;
+
+public class AmazonProductService extends AbstractAPIHandler implements APIHandler {
+
+	private static final long serialVersionUID = 2279773523424505716L;
+
+	// keys available on request
+	private static final String AWS_ACCESS_KEY_ID = "";
+	private static final String AWS_SECRET_KEY = "";
+	private static final String ASSOCIATE_TAG = "";
+
+	// using the USA locale
+	private static final String ENDPOINT = "webservices.amazon.com";
+
+	@Override
+	public String getAPIPath() {
+		return "/cms/amazonservice.json";
+	}
+
+	@Override
+	public BaseUserRole getMinimalBaseUserRole() {
+		return BaseUserRole.ANONYMOUS;
+	}
+
+	@Override
+	public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
+		return null;
+	}
+
+	public static JSONObject fetchResults(String requestUrl, String operation) {
+		JSONObject itemlookup = new JSONObject(true);
+		try {
+			DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+			DocumentBuilder db = dbf.newDocumentBuilder();
+			Document doc = db.parse(requestUrl);
+			DOMSource domSource = new DOMSource(doc);
+			StringWriter writer = new StringWriter();
+			StreamResult result = new StreamResult(writer);
+			TransformerFactory tf = TransformerFactory.newInstance();
+			Transformer transformer = tf.newTransformer();
+			transformer.transform(domSource, result);
+			JSONObject xmlresult = new JSONObject(true);
+			xmlresult = XML.toJSONObject(writer.toString());
+			JSONObject items = xmlresult.getJSONObject(operation).getJSONObject("Items");
+			if (items.getJSONObject("Request").has("Errors")) {
+				itemlookup.put("status", "error");
+				itemlookup.put("reason",
+						items.getJSONObject("Request").getJSONObject("Errors").getJSONObject("Error").get("Message"));
```

Comment:
Yes. I am implementing the ItemLookup and the ItemSearch API, and they are always in this format.

There are other APIs too: like SimilarityLookup etc, but the result of SimilarityLookup (and all the other APIs) comes when ```response_group=Large```, and as I wrote it's that by default in my code, so if you want all the results, just fetch an ```id``` or a ```q``` and leave response_group blank.